### ISSUE TITLE
[WiKi] How to Build clawPDF MSI yourself?

### ISSUE THREAD

**Issue by hansaliyad1** [clawsoftware/clawPDF#135]
I am not sure if this is the correct way to do it, but it works. I was exploring this repo for the past two days to figure out how to build my clawPDF fork but did not find an answer. This is for anyone who is looking to build their own clawPDF fork. Feel free to correct me if this is not the correct way to build this project.

Before we build our MSI, we need to add our own code signing certificate or remove the existing code signing certificate configuration. We have to do this for the clawPDF, clawPDF.Bridge, clawPDF.Core, clawPDF.Mail, clawPDF.Settings, clawPDF.Shared, clawPDF.Utilities, ftplib, PDFProcessing, and SetupHelper projects. Right-click on each of these projects, click on "Properties", click on "Signing", and then under "Sign the assembly", either uncheck it or add your own code signing certificate.

Now, how do we build our MSI?

1. Set the Build Configuration to Release -> Any CPU -> clawPDF.
2. Right-click on the solution "clawPDF" and then click on "Build Solution".
3. Change the Build Configuration to Release -> Win32 -> clawPDF.
4. Right-click on the project "PrinterMonitor" and then click on "Build".
5. Change the Build Configuration to Release -> ARM64 -> clawPDF.
6. Right-click on the project "PrinterMonitor" and then click on "Build".
7. Right-click on the project "clawPDF_setup" and then click on "Build".
8. You should now find your built MSI under "{PROJECT_DIR}\clawPDF\src\clawPDF_setup\Release".

If anyone has any questions, please feel free to comment below. Thank you!

**Comment by akzkorosh** [comment]
Thank you very much for your guidelines. Do you have any idea about resolving issue with that missing signClawPDF.pfx

**Comment by hansaliyad1** [comment]
signClawPDF.pfx is a certificate that is used by clawsoftware to sign and release the clawPDF program. Like I mention in my post, we need to add our own code signing certificate or remove the existing code signing certificate configuration. We have to do this for the clawPDF, clawPDF.Bridge, clawPDF.Core, clawPDF.Mail, clawPDF.Settings, clawPDF.Shared, clawPDF.Utilities, ftplib, PDFProcessing, and SetupHelper projects. Right-click on each of these projects, click on "Properties", click on "Signing", and then under "Sign the assembly", either uncheck it or add your own code signing certificate.

From: akzkorosh ***@***.***>
Date: Thursday, May 30, 2024 at 6:09 PM
To: clawsoftware/clawPDF ***@***.***>
Cc: Dharmang Hansaliya ***@***.***>, Author ***@***.***>
Subject: Re: [clawsoftware/clawPDF] [WiKi] How to Build clawPDF MSI yourself? (Issue #135)

Thank you very much for your guidelines. Do you have any idea about resolving issue with that missing signClawPDF.pfx

**Comment by hueldoeu** [comment]
what kind of software do you use? visual studio?

**Comment by hansaliyad1** [comment]
@hueldoeu Yes, I use Visual Studio to build the project. Sometimes, I also use Jetbrains Rider but you cannot build this project using that.

**Comment by Luolinxiong** [comment]
You need to add some procedure with the Visual Studio 2022 Update and clawPDF update.

1. Add procedure after step 6 : Change the Build Configuration to Release -> x64 -> clawPDF.
2. Right-click on the project "PrinterMonitor" and then click on "Build".
If you build filed after step 7,you need to observe the error information because you must omission some project to build (like clawPDF.Bridge, clawPDF.Core, clawPDF.Mail, clawPDF.Settings, clawPDF.Shared, clawPDF.Utilities, ftplib, PDFProcessing, and SetupHelper projects)
3. Right-click on the project according to  your error information and then click on "Build".
4. Again right-click on the project "clawPDF_setup" and then click on "Build".
You will find it is success to have a msi/exe

**Comment by Luolinxiong** [comment]
There are some thing was changed with the update of Visual Studio 2022, you have to build step 6(PrinterMonitor) with four ways Any CPU ARM64 Win32 and X64 together.
